src/librustc/hir/lowering.rs

```diff
@@ -1029,6 +1035,20 @@ impl<'a> LoweringContext<'a> {
         result
     }
 
+    fn with_dyn_type_scope<T, F>(&mut self, in_scope: bool, f: F) -> T
```

Comment:
This idiom occurs a few times in this file, could you try:
```rust
fn with_scope<S, T>(&mut self, store: &mut S, with: S, f: impl FnOnce(&mut Self) -> T) -> T {
    let old = mem::replace(store, with);
    let result = f(self);
    *store = old;
    result
}
```
as an implementation of this function?

Reply:
I think it's cleaner as it is presently.